src/concourse/lib/jobs/infrastructure.py

```diff
@@ -170,13 +175,24 @@ def pulumi_jobs_chain(
             # These mutations apply globally if the dependencies aren't copied below
             dependency.trigger = not bool(previous_job or production_stack)
             dependency.passed = passed_param or dependency.passed  # type: ignore
+
+        # Need to copy the dependencies because otherwise they are globally mutated
+        local_dependencies = [
+            dependency_step.copy() for dependency_step in (dependencies or [])
+        ]
+        if index in (custom_dependencies or {}):
```

Comment:
You could probably even make this a little cleaner like this:
```
if custom_dependency := (custom_dependencies or {}).get(index):
    local_custom_dependencies = [custom_dependency_step.copy() for custom_dependency_step in custom_dependency]
```

Reply:
That uses the "walrus operator" also known as conditional assignment. That allows you to avoid repeating the attribute access.